boa/src/builtins/object/internal_methods.rs

```diff
@@ -282,25 +280,162 @@ pub trait ObjectInternalMethods {
         true
     }
 
-    /// https://tc39.es/ecma262/#sec-ordinary-object-internal-methods-and-internal-slots-getownproperty-p
-    /// The specification returns a Property Descriptor or Undefined. These are 2 separate types and we can't do that here.
-    fn get_own_property(&self, prop: &Value) -> Property;
+    /// The specification returns a Property Descriptor or Undefined.
+    ///
+    /// These are 2 separate types and we can't do that here.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-ordinary-object-internal-methods-and-internal-slots-getownproperty-p
+    pub fn get_own_property(&self, prop: &Value) -> Property {
+        let _timer = BoaProfiler::global().start_event("Object::get_own_property", "object");
+
+        debug_assert!(Property::is_property_key(prop));
+        // Prop could either be a String or Symbol
+        match *(*prop) {
+            ValueData::String(ref st) => {
+                match self.properties.get(st) {
+                    // If O does not have an own property with key P, return undefined.
+                    // In this case we return a new empty Property
+                    None => Property::default(),
+                    Some(ref v) => {
+                        let mut d = Property::default();
+                        if v.is_data_descriptor() {
+                            d.value = v.value.clone();
+                            d.writable = v.writable;
+                        } else {
+                            debug_assert!(v.is_accessor_descriptor());
+                            d.get = v.get.clone();
+                            d.set = v.set.clone();
+                        }
+                        d.enumerable = v.enumerable;
+                        d.configurable = v.configurable;
+                        d
+                    }
```

Comment:
Do you think a `map(||...).unwrap_or_else(Property::default)` would be easier to understand? or an `if let`?

Reply:
```rust
self.properties().get(st).map_or_else(Property::default, || {
	// ...
})
```
This is with `map_or_else`

Reply:
We should do this, right?

Reply:
sorry forgot about it :sweat_smile: . I'll do it now.